jdbc-driver/src/main/java/com/google/cloud/dataproc/jdbc/DataprocDriver.java

```diff
@@ -71,12 +74,26 @@ public Logger getParentLogger() throws SQLFeatureNotSupportedException {
      * jdbc:hive2://<host>:<port>/<dbName>;transportMode=http;httpPath=<http_endpoint>;
      * <otherSessionConfs>?<hiveConfs>#<hiveVars>
      *
-     * @param url user passed in JDBC URL
-     * @param info user passed in connection properties
+     * @param url client passed in JDBC URL
+     * @param info client passed in connection properties
      * @return the created Hive Connection
      */
     @VisibleForTesting
-    Connection createConnection(String url, Properties info) {
-        return null;
+    Connection createConnection(String url, Properties info) throws SQLException {
+        // Valid url format:
+        // jdbc:dataproc://<protocol>/<db>;clusterName=<>;other_sess_var_list?hive_conf_list#hive_var_list
+
+        if (Pattern.matches(DATAPROC_JDBC_HIVE_URL_SCHEMA + ".*", url)) {
+            try {
+                String hiveURL = formatHiveUrl(url);
+            } catch (InvalidURLException e) {
+                throw new SQLException();
```

Comment:
Add `e` as a cause of `SQLException`  to not lose stack trace and simplify debugging.